frontend/catalyst/device/verification.py

```diff
@@ -209,7 +209,7 @@ def _op_checker(op, state):
         # is handled in _inv_op_checker and _ctrl_op_checker.
         # Specialed control op classes (e.g. CRZ) should be checked directly though, which is why we
         # can't use isinstance(op, Controlled).
-        if type(op) in (Controlled, ControlledOp) or isinstance(op, Adjoint):
+        if type(op) in (Controlled, ControlledOp) or isinstance(op, (Adjoint, StatePrepBase)):
```

Comment:
I should have updated the comment. This line is just a special case to avoid verification and since StatePrep is not on the list of accepted devices it should be skipped as well.

Reply:
Still logically it does not belong here, better to skip it in its own branch.

Reply:
But also, why are we just skipping verification? Shouldn't it confirm that the device allows it?

Reply:
We agreed that StatePrep would not appear in the TOML file. The only reason why it is here is because it is already in the `initial_state_prep_flag=True`. So, it is here because the device allows it, and you want me to verify if it is here that the device allows it?

Reply:
> So, it is here because the device allows it, and you want me to verify if it is here that the device allows it?

Why the gate is present at this stage we don't know, since the pre-processing can get complicated (and potentially customized) the point of the verifier is that it is run independently of any pre-processing that comes before it, so it should indeed check that the device supports it.

Reply:
https://github.com/PennyLaneAI/catalyst/pull/955/commits/deeaf15e4f1fe7b83c22b2a0213fa4b56d79f7be